zerver/actions/users.py

```diff
@@ -512,7 +519,13 @@ def do_deactivate_user(
         # in only the user being deactivated.
         bot_profiles = get_active_bots_owned_by_user(user_profile)
         for profile in bot_profiles:
-            do_deactivate_user(profile, _cascade=False, acting_user=acting_user)
+            do_deactivate_user(
+                profile,
+                _cascade=False,
+                acting_user=acting_user,
+                is_spammer=is_spammer,
+                message_delete_action=message_delete_action,
```

Comment:
This seems fine to consider bot as spammer if their owner is spammer and take same action on messages, but should we also do the things like renaming them or removing their profile picture, etc? Is there some discussion on how to handle bots owned by spammer? If not would be good to discuss it once.